.changeset/small-ties-sort.md

```diff
@@ -0,0 +1,50 @@
+---
+'astro': major
+---
+
+Fixes attribute rendering for non-[boolean HTML attributes](https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML) with boolean values to match proper attribute handling in browsers.
+
+Previously, non-boolean attributes may not have included their values when rendered to HTML. In Astro v5.0, the values are now explicitly rendered as `="true"` or `="false"`
+
+In the following `.astro` examples passed values of `true` and `false`, only `allowfullscreen` is a boolean attribute:
+
+```astro
+<!-- src/pages/index.astro -->
+<!-- `allowfullscreen` is a boolean attribute -->
+<p allowfullscreen={true}></p>
+<p allowfullscreen={false}></p>
+
+<!-- `inherit` is *not* a boolean attribute -->
+<p inherit={true}></p>
+<p inherit={false}></p>
+
+<!-- `data-*` attributes are not boolean attributes -->
+<p data-light={true}></p>
+<p data-light={false}></p>
+```
+
+Astro v5.0 now preserves the full data attribute with its value when rendering the HTML of non-boolean attributes:
+
+```diff
+  <p allowfullscren></p>
+  <p></p>
+
+  <p unknown="true"></p>
+- <p unknown></p>
++ <p unknown="false"></p>
+
+- <p data-foo></p>
++ <p data-foo="true"></p>
+- <p></p>
++ <p data-foo="false"></p>
```

Comment:
Since you have renamed the attributes for lines 13 to 23 [in this review comment](https://github.com/withastro/astro/pull/11660#discussion_r1711519745), I think we have an inconsistency. This should be:
```suggestion
  <p inherit="true"></p>
- <p inherit></p>
+ <p inherit="false"></p>

- <p data-light></p>
+ <p data-light="true"></p>
- <p></p>
+ <p data-light="false"></p>
```